embedded-graphics/src/primitives/triangle/mod.rs

```diff
@@ -185,10 +205,14 @@ impl Triangle {
     /// Point inside triangle, ignoring pixels that partially lie outside triangle lines.
     pub(self) fn mathematical_contains(&self, point: &Point) -> bool {
         // Skip expensive calculations below if point is outside the bounding box
-        if !self.bounding_box().contains(*point) {
-            return false;
+        if self.bounding_box().contains(*point) {
+            self.in_mathematical_triangle(point)
+        } else {
+            false
         }
```

Comment:
This could be condensed using Rust's short circuiting rules:

```suggestion
        self.bounding_box().contains(*point) && in_mathematical_triangle(point)
```

Reply:
I didn't want to go fancy :)